File: app/src/main/java/apps/jizzu/simpletodo/alarm/AlarmSetter.java
package apps.jizzu.simpletodo.alarm;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import java.util.ArrayList;
import java.util.List;

import apps.jizzu.simpletodo.database.DBHelper;
import apps.jizzu.simpletodo.model.ModelTask;

/**
 * Class for restoring all notifications after device reboot.
 */
public class AlarmSetter extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        DBHelper helper = DBHelper.getInstance(context);

        AlarmHelper.getInstance().init(context);
        AlarmHelper alarmHelper = AlarmHelper.getInstance();

        List<ModelTask> tasks = new ArrayList<>();
        tasks.addAll(helper.getAllTasks());

        for (ModelTask task : tasks) {
            if (task.getDate() != 0) {
                alarmHelper.setAlarm(task);
            }
        }
    }
}

File: app/src/main/java/apps/jizzu/simpletodo/alarm/AlarmSetter.kt
package apps.jizzu.simpletodo.alarm

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

import java.util.ArrayList

import apps.jizzu.simpletodo.database.DBHelper
import apps.jizzu.simpletodo.model.ModelTask

/**
 * Class for restoring all notifications after device reboot.
 */
class AlarmSetter : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val helper = DBHelper.getInstance(context)

        AlarmHelper.getInstance().init(context)
        val alarmHelper = AlarmHelper.getInstance()

        val tasks = ArrayList<ModelTask>()
        tasks.addAll(helper.getAllTasks())

        for (task in tasks) {
            if (task.date != 0L) {
                alarmHelper.setAlarm(task)
            }
        }
    }
}
